### DISCUSSION TITLE
Question on Evaluation of Long Expressions with FermionSpinSum

### DISCUSSION THREAD

**Discussion by ghost** [FeynCalc/feyncalc#196] [Q&A] [answered]
Hi,

I am trying to calculate a 2->3 body Feynman diagram at tree level. The amplitude contains a Rarita Schwinger propagator, a Proca Propagator, three couplings and 4 spinors:

`M[1]= SpinorUBar[p1,m1] . Coupling[ρ,N,N, pb-p1,μ] .  SpinorU [pb,mb] . NumPropPro[pb-p1,m[ρ],μ,ν] . SpinorUBar[p2,m2] .  Coupling[k,Λ,N1720, p3,σ] . NumPropRS[pa+pb-p1,m[1720],σ,τ] . Coupling[ρ,N,N1720, τ,ν]  . SpinorU [pa,ma];`

with:
 
`Coupling[ρ,N,N,  mesonmomentum_,lorentzindex_]:=-I* g[ρ,N,N]*( GA[lorentzindex] -I*κ/(2*m[N])DiracSigma[ GS[mesonmomentum],GA[lorentzindex]] ) `

`Coupling[ρ,N,N1720, lorentzindexmeson_,lorentzindexrs_]:=-I* g[ρ,N,N1720]*GA[5] . MT[lorentzindexmeson,lorentzindexrs] 
`
 
`Coupling[k,Λ,N1720, mesonmomentum_,lorentzindex_]:= g[k,Λ,N1720]/m[k]* FV[mesonmomentum,lorentzindex]  
`

`NumPropRS[p_,m_,μ_,ν_]:=(GS[p]+m) . (MT[μ,ν]-1/3 GA[μ] . GA[ν]-1/(3m)*(GA[μ] . FV[p,ν]-GA[ν] . FV[p,μ])-2/(3m^2)FV[p,μ] . FV[p,ν]);`

`NumPropPro[p_,m_,μ_,ν_]:=(MT[μ,ν]-FV[p,μ] . FV[p,ν]/m^2)`

For evaluating such an expression i use:

`Numer[Amp_,AmpConj_]:=Module[{A=Amp,B=AmpConj,fersumed,fersumed2,fersumed3,result}, ` 
`fersumed=(FermionSpinSum[#1,ExtraFactor->1/2^2]&)[ A*ComplexConjugate[B]]//DiracSimplify;`
`fersumed2=fersumed//.ScalProd;`
`result=fersumed2//.DotPro;   `
`result`
`]`

where the two rules are for inserting the kinematics(thus irrelevant for the discussion). My question is what is the correct way to write the amplitude in order to be evaluated by `FermionSpinSum` . This way gives a result,but the result is wrong.

Another Amplitude that fails to give the correct result but is evaluated must faster is:

`M[2]= SpinorUBar[p1,m1] . Coupling[ρ,N,N,-pa+p1,μ] .  SpinorU [pa,ma]  .  NumPropPro[pb-p1,m[ρ],μ,ν] . SpinorUBar[p2,m2] . Coupling[k,Λ,N1710]  . NumPropFer[pa+pb-p1,m[1710]]  .  Coupling[ρ,N,N1710, ν]  . SpinorU [pb,mb] ;`

with:

`Coupling[k,Λ,N1710 ]:=  g[k,Λ,N1710]*GA[5] `
`Coupling[ρ,N,N1710,lorentzindex_ ]:=-I g[ρ,N,N1710]* GA[lorentzindex] `

This code part is used to recreate an already known result though my question is on what the correct syntax is. I have used the same `Numer` function and same procedures on amlitudes that are of the same form but with a non-vector coupling on the first term (i.e. `SpinorUBar[p1,m1] . ( GA[5]  ) .  SpinorU [pa,ma]`)  and with a Scalar propagator instead of Proca and the results are exactly like those I was looking for. Any suggestions on what could be the problem?

P.S. I am using mathematica 12.2 and FeynCalc 9.3.1

**Comment by vsht** [comment, marked as answer]
Hi,

this looks like a quite complicated expression with lots of custom functions. 

I don't see any immediate issues with it, but if it doesn't evaluate to what you expect, you'll have to disentangle it into small blocks (e.g. using `DotSimplify` and `Collect2`) and check them one by one until you find the error (or a bug).

I would also evaluate it directly (not inside a Module) to avoid possible variable shadowings.

**Reply by ghost** [reply]
Hi,

Thank you for the reply and suggestions. I have looked more into it and seems that the `FermionSpinSum` expression evaluates correctly. 

This is a part of a larger code, so I guess the problem hides somewhere else.

Thank you for your time,
K.P.
